hazelcast/src/main/java/com/hazelcast/nio/serialization/ClassDefinitionBuilder.java

```diff
@@ -183,58 +167,71 @@ public ClassDefinitionBuilder addFloatField(String fieldName) {
      * @throws HazelcastSerializationException if a field with same name already exists or
      *                                         if this method is called after {@link ClassDefinitionBuilder#build()}
      */
-    public ClassDefinitionBuilder addShortField(String fieldName) {
-        check(fieldName);
-        fieldDefinitions.add(new FieldDefinitionImpl(index++, fieldName, FieldType.SHORT, version));
-        return this;
+    public ClassDefinitionBuilder addShortField(@Nonnull String fieldName) {
+        return addField(fieldName, FieldType.SHORT);
     }
 
     /**
+     * Adds a decimal which is arbitrary precision and scale floating-point number to the class definition
+     *
      * @param fieldName name of the field that will be added to this class definition
      * @return itself for chaining
      * @throws HazelcastSerializationException if a field with same name already exists or
      *                                         if this method is called after {@link ClassDefinitionBuilder#build()}
      */
-    public ClassDefinitionBuilder addByteArrayField(String fieldName) {
-        check(fieldName);
-        fieldDefinitions.add(new FieldDefinitionImpl(index++, fieldName, FieldType.BYTE_ARRAY, version));
-        return this;
+    public ClassDefinitionBuilder addDecimalField(@Nonnull String fieldName) {
+        return addField(fieldName, FieldType.DECIMAL);
     }
 
     /**
+     * Adds a time field consisting of hour, minute, seconds and nanos parts to the class definition
```

Comment:
The idea is that we may support multiple types in the future. 
I wanted to make what is carries as data obvious so that user will know what to expect when we support multiple types.

The supported types will be clear on their respective read/write methods on PortableReader/Writer from the parameters&return types.

Reply:
In this case we should at least reference corresponding methods of PortableReader/Writer. Current javadoc leaves the reader with too many questions.